### ISSUE TITLE
Add backwards-compatible message passing for legacy apps

### ISSUE THREAD

**Issue by hstove** [leather-io/extension#937]
After #930 , apps with older versions of Connect won't be able to utilize these APIs. We can still add support for the legacy message passing behavior. It would be nice to do this - otherwise you couldn't login to older apps with the web wallet.

**Comment by friedger** [comment]
Is it worth the effort? How many older apps are still used and maintained?

**Comment by hstove** [comment]
Good question. I mean this is an issue if someone wants to log in to apps like Blocksurvey with the web wallet. I think one problem right now is that the injected `BlockstackProvider` variable is used by all Connect apps, and when it sees that, it tries to authenticate with the extension. And that doesn't work, now. A simpler solution is to remove the `BlockstackProvider` injected variable - it's been renamed to `StacksProvider` now, anyways. Removing the alias for `BlockstackProvider` will allow non-stacks-2 apps to go straight to app.blockstack.org.

**Comment by markmhendrickson** [comment]
This seems to address problems for the venn diagram of users that have both A) used apps like BlockSurvey who haven't upgraded to the new Stacks Wallet but rather still rely on app.blockstack.org, and B) have separately installed the new Stacks Wallet (either directly for via a second app).

My sense is that this is a small subset of rather savvy users who could otherwise just disable the extension while using apps like BlockSurvey. So we may want to keep this in the backlog until we hear otherwise.

**Comment by dantrevino** [comment]
Runkod, NoteRiot (for now), LandHo, ClarityTools ... are all impacted ...  I'm sure I can find more ...

**Comment by hstove** [comment]
Yes. They also have that same unfortunate Firefox bug. Ultimately updating their version of `@stacks/connect` is necessary to fix that, at least for transaction apps like ClarityTools.

I'm leaning towards not intercepting the auth request for apps that don't use transaction signing
